text/003-trigger-authority.md

```diff
@@ -0,0 +1,134 @@
+| By-call                                                                    | Use caller's (who executed the `ExecuteTrigger` instruction) |
+| Also, if any of the data triggers fail, the entire transaction also fails. |                                                              |
+
+## Motivation to change
+
+### Use case 1 - deploying on-chain functions callable by less-privileges users but executed with higher privileges
+
+While building the [Hub Chain PoC](https://github.com/hyperledger-iroha/hub-chain-poc), we try to design an on-chain
+agent that would accept an input from an untrusted external entity (a relay), validate it, and perform certain
+administrative actions[^2].
+
+[^2]: Specifically, the on-chain agent would receive block headers of another chain and proofs that e.g. a transfer
+    happened there. Then, if the on-chain agent can prove, it would make e.g. a local transfer.
+
+It would be convenient to register a by-call trigger that would accept transactions from a relay in a form of
+`ExecuteTrigger { id: "our-special-trigger", args: ... }`, validate, and perform administrative actions. However, in the
+current design the trigger would infer the authority of the caller, i.e. of the relay's account. Therefore, we would
+have to give the relay's account sufficient permissions for the trigger to perform these actions. In this case, the
+relay could perform these action by itself without an on-chain trust-less validation, which defeats the purpose.
+
+#### Alternatives
+
+- **Use Pipeline/Time filters**. These triggers are always executed using the `authority` set in the trigger. Therefore,
+  they can have administrative permissions, react on some state changes made by an untrusted source, and execute
+  administrative actions.
+- **Use Trigger Completed filter**. Then, an un-trusted source could call some un-trusted trigger that would create some
+  state change, and then the administrative trigger would execute as a result of that trigger completion, performing
+  actions with elevated permissions. This works essentially the same as the Pipeline/Time variant, but with immediate
+  execution.
+- **Use executor-level custom instructions**. Executor has "root privileges" and can perform any action. However, it is
+  not a modular and composable approach, as there is only one global executor in Iroha. With triggers, it is possible
+  for different unrelated authorities to deploy their own permission-ed logic without interfering with each other. It's
+  just more modular, come on!
+
+### Use case 2
+
+I (e.g. Alice) want to deploy a data trigger that reacts on someone else's (e.g. Bob's) asset balance change. This
+trigger would, for example, make a transfer on _my_ behalf.
+
+However, when Bob changes his own balance, my trigger would be executed with _Bob's_ permissions, not mine, and
+therefore the trigger will fail to make the intended transfer.
```

Comment:
As an alternative, Alice could keep a multisig proposal pending and have it approved by a data trigger, which would achieve the same effect.

Reply:
Doesn't it sound kind of comlicated?